app/controllers/account/registrations_controller.rb

```diff
@@ -34,6 +35,26 @@ def destroy
       not_found
     end
 
+    # rubocop:disable Metrics/AbcSize
+    def phone_verification
```

Comment:
Test etmeden yazdığım aşağıdaki kod bana bir parça daha okunur geliyor (lojiği biraz daha iyi takip edebiliyorum), ama bu subjektif olabilir.  Nihai kararı tamamen sana bırakıyorum (comment olarak yazdım), sıkıntı yok :-)  Not: Bu kodun ABC kompleksitesi biraz daha fazla: 15.81/15, ama amacım bunu düşürmek değil.

```ruby
def phone_verification
  phone = current_user[:mobile_phone] = params.dig(:user, :mobile_phone)

  return respond_to(:js) unless current_user.valid?
  return respond_to(:js) unless current_user.mobile_phone_changed?

  return respond_to(:js) if Twilio::Verify.send_phone_verification_code(phone) == 'ok'

  respond_to do |format|
    format.js { flash[:alert] = t('errors.system_error') }
  end
end

```

Reply:
Burasını da bu şekilde yapabiliriz ama tam test etmedim.

```ruby
def phone_verification
  current_user.assign_attributes(params_for_phone_update)

  if current_user.valid? && current_user.mobile_phone_changed?
    flash[:alert] = t('errors.system_error') unless Twilio::Verify.send_phone_verification_code(params_for_phone_update[:moible_phone]) == 'ok'
  end

  respond_to :js
end

private

def params_for_phone_update
  params.require(:user).permit(:mobile_phone)
end
```

Reply:
Veya böyle de oluyor sanırım:

```ruby
def phone_verification
  phone = current_user[:mobile_phone] = params.dig(:user, :mobile_phone)

  return respond_to(:js) unless current_user.valid?
  return respond_to(:js) unless current_user.mobile_phone_changed?

  flash[:alert] = t('errors.system_error') unless Twilio::Verify.send_phone_verification_code(phone) == 'ok'

  respond_to :js
end
```

Not: Kişisel düşüncem

```ruby
if expr1 && expr3
  do_something if expr3
end
```

şeklinde kodları tercih etmemeye eğilimliyim.  :-)  Yukarıda önerdiğim düzenleme de bunu yansıtıyor.  Ama bu böyle çok fazla temellendirdiğim/arkasında durduğum bir konu değil.  Önemli olan tutarlılık.  Örneğin önerdiğim şekilde gidilecekse bu patern PR'daki diğer kodlarda da var, onlar da değişmeli.  Aksi halde önerimi göz ardı edebilirsin, çünkü tutarlılık daha önemli